### DISCUSSION TITLE
JUnit to CTRF converter

### DISCUSSION THREAD

**Discussion by timkrins** [ctrf-io/ctrf#8] [Reporters]
Hello,
This is a brilliant idea.
To speed up adoption, I wonder if it is a good idea to prioritise a JUnit converter, as so many tools already have JUnit compatible exporters.
For example, I currently have a pipeline that uses many different test frameworks, but all output JUnit XML, and now I want to convert these results into something more agnostic for formatting.
I know the conversion output may be missing all the context that the CTRF spec may provide, but something for now is better than nothing!

**Comment by Ma11hewThomas** [comment]
Thank you for the feedback; it's a great idea.

I think a good approach would be to write a Node CLI that users can call in their workflows, e.g npx convert-junit-to-ctrf

I've added [junit-ctrf-convert](https://github.com/ctrf-io/junit-ctrf-convert) with a ticket to write the initial implementation.

**Comment by Ma11hewThomas** [comment]
See https://github.com/ctrf-io/junit-to-ctrf

**Reply by timkrins** [reply]
Excellent! Speedy work!
